libraries/c_sdk/standard/mqtt/test/unit/iot_tests_mqtt_subscription.c

```diff
@@ -261,6 +270,80 @@ static void _blockingCallback( void * pArgument,
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Setting the MQTT Context for the given MQTT Connection.
+ *
+ */
+static void _setContext( IotMqttConnection_t pMqttConnection )
+{
+    bool subscriptionMutexCreated = false;
+    bool contextMutex = false;
+    TransportInterface_t transport;
+    MQTTFixedBuffer_t networkBuffer;
+    MQTTApplicationCallbacks_t callbacks;
+    MQTTStatus_t managedMqttStatus = MQTTBadParameter;
+    IotMqttError_t status = IOT_MQTT_BAD_PARAMETER;
+
+    /* Getting the free index from the MQTT connection to MQTT context mapping array. */
+    contextIndex = _IotMqtt_getFreeIndexFromContextConnectionArray();
+
+    /* Creating Mutex for the synchronization of MQTT Context used for sending the packets
+     * on the network using MQTT LTS API. */
+    contextMutex = IotMutex_CreateRecursiveMutex( &( connToContext[ contextIndex ].contextMutex ) );
+
+    /* Create the subscription mutex for a new connection. */
+
+    if( contextMutex == true )
+    {
+        /* Assigning the MQTT Connection. */
+        connToContext[ contextIndex ].mqttConnection = pMqttConnection;
+
+        subscriptionMutexCreated = IotMutex_CreateNonRecursiveMutex( &( connToContext[ contextIndex ].subscriptionMutex ) );
+
+        if( subscriptionMutexCreated == false )
+        {
+            IotLogError( "Failed to create subscription mutex for new connection." );
```

Comment:
We cannot proceed with a failure in this step

Reply:
Addressed